repositories/src/test/scala/cromwell/pipeline/datastorage/dao/repository/ProjectRepositoryTest.scala

```diff
@@ -1,33 +1,22 @@
 package cromwell.pipeline.datastorage.dao.repository
 
-import com.dimafeng.testcontainers.{ ForAllTestContainer, PostgreSQLContainer }
-import com.typesafe.config.Config
-import cromwell.pipeline.datastorage.DatastorageModule
-import cromwell.pipeline.datastorage.dao.utils.{ TestProjectUtils, TestUserUtils }
+import cromwell.pipeline.datastorage.dao.utils.{ PostgreTestSources, TestProjectUtils, TestUserUtils }
 import cromwell.pipeline.datastorage.dto.{ PipelineVersion, Project, User }
-import cromwell.pipeline.utils.{ ApplicationConfig, TestContainersUtils }
-import org.scalatest.{ AsyncWordSpec, BeforeAndAfterAll, Matchers }
+import org.scalatest.Matchers
 
-class ProjectRepositoryTest extends AsyncWordSpec with Matchers with BeforeAndAfterAll with ForAllTestContainer {
+class ProjectRepositoryTest extends PostgreTestSources with Matchers {
 
-  override val container: PostgreSQLContainer = TestContainersUtils.getPostgreSQLContainer()
-  private lazy val config: Config = TestContainersUtils.getConfigForPgContainer(container)
-  private lazy val datastorageModule: DatastorageModule = new DatastorageModule(ApplicationConfig.load(config))
+  container.start()
 
-  override protected def beforeAll(): Unit = {
-    super.beforeAll
-    datastorageModule.pipelineDatabaseEngine.updateSchema()
-  }
-
-  private val dummyUser: User = TestUserUtils.getDummyUser()
-  private val dummyProject: Project = TestProjectUtils.getDummyProject(ownerId = dummyUser.userId)
+  import datastorageModule._
 
-  import datastorageModule.{ projectRepository, userRepository }
   "ProjectRepository" when {
 
     "getUserById" should {
 
       "find newly added project by id" taggedAs Dao in {
+        val dummyUser: User = TestUserUtils.getDummyUser()
+        val dummyProject: Project = TestProjectUtils.getDummyProject(ownerId = dummyUser.userId)
```

Comment:
Are we realy need to add User and Project every time when we start new step ?
I think we can write this after " should {" only once.

Reply:
Thats a good question! As I can see in some of our Repository tests different approach is used. For example in both UserRepositoryTest and RunRepositoryTest, tested entities are created just before every testcase. I doubt how should I handle this as both approaches work. Creating new dummies in every testcase may be better since then we have more autonomous independent cases. On the other hand, since we truncate tables after each testcase, theoretically we could add dummies as lazy vals to a new trait and use one dummy ref for all repository tests.

Reply:
Thats a good question! As I can see in some of our Repository tests different approach is used. For example in both UserRepositoryTest and RunRepositoryTest, tested entities are created just before every testcase. I doubt how should I handle this as both approaches work. Creating new dummies in every testcase may be better since then we have more autonomous independent cases. On the other hand, since we truncate tables after each testcase, theoretically we could add dummies as lazy vals to a new trait and use one dummy ref for all repository tests.

Reply:
Agree with Pavel, in these cases we are testing functions, not data.

Reply:
Fixed